roachtest: activerecord failed

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2988804&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2988804&tab=artifacts#/activerecord) on master @ [25e26a5f0361fcfbf71cee86f963b93f806604d8](https://github.com/cockroachdb/cockroach/commits/25e26a5f0361fcfbf71cee86f963b93f806604d8):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/activerecord/run_1
	orm_helpers.go:245,orm_helpers.go:171,activerecord.go:229,activerecord.go:241,test_runner.go:784: 
		Tests run on Cockroach v21.2.0-alpha.00000000-542-g25e26a5f03
		Tests run against activerecord 6.1
		6587 Total Tests Run
		6585 tests passed
		2 tests failed
		18 tests skipped
		1 test ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: IgnoreFixturesTest#test_ignores_books_fixtures (unexpected)
		--- FAIL: ActiveRecord::Migration::ForeignKeyTest#test_add_foreign_key_with_non_standard_primary_key (unexpected)
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList21_2) is available in the artifacts' activerecord log
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh activerecord
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #65297 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-21.1]
- #65157 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-20.2]
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2998287&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2998287&tab=artifacts#/activerecord) on master @ [c305dbcff1bcf621ae8edf058b220033c98d6526](https://github.com/cockroachdb/cockroach/commits/c305dbcff1bcf621ae8edf058b220033c98d6526):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/activerecord/run_1
	orm_helpers.go:245,orm_helpers.go:171,activerecord.go:229,activerecord.go:241,test_runner.go:792: 
		Tests run on Cockroach v21.2.0-alpha.00000000-626-gc305dbcff1
		Tests run against activerecord 6.1
		6587 Total Tests Run
		6584 tests passed
		3 tests failed
		18 tests skipped
		1 test ignored
		0 tests passed unexpectedly
		3 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: ActiveRecord::Migration::ReferencesForeignKeyInCreateTest#test_foreign_keys_can_be_created_with_the_table (unexpected)
		--- FAIL: IgnoreFixturesTest#test_ignores_books_fixtures (unexpected)
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: ActiveRecord::WhereChainTest#test_not_eq_with_succeeding_where (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList21_2) is available in the artifacts' activerecord log
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh activerecord
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #65297 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-21.1]
- #65157 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-20.2]
</p>
</details>
/cc cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3031655&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3031655&tab=artifacts#/activerecord) on master @ [995f0dd1512d0266e78001980aab0023eab0e70f](https://github.com/cockroachdb/cockroach/commits/995f0dd1512d0266e78001980aab0023eab0e70f):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/activerecord/run_1
	orm_helpers.go:245,orm_helpers.go:171,activerecord.go:229,activerecord.go:241,test_runner.go:792: 
		Tests run on Cockroach v21.2.0-alpha.00000000-899-g995f0dd151
		Tests run against activerecord 6.1
		6587 Total Tests Run
		6586 tests passed
		1 test failed
		18 tests skipped
		1 test ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: IgnoreFixturesTest#test_ignores_parrots_fixtures (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList21_2) is available in the artifacts' activerecord log
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh activerecord
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #65297 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-21.1]
- #65157 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-20.2]
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3043009&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3043009&tab=artifacts#/activerecord) on master @ [7869b781b66823c44f1b30047d0fcbcc2dbe8e0f](https://github.com/cockroachdb/cockroach/commits/7869b781b66823c44f1b30047d0fcbcc2dbe8e0f):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/activerecord/run_1
	orm_helpers.go:245,orm_helpers.go:171,activerecord.go:229,activerecord.go:241,test_runner.go:801: 
		Tests run on Cockroach v21.2.0-alpha.00000000-1138-g7869b781b6
		Tests run against activerecord 6.1
		6587 Total Tests Run
		6586 tests passed
		1 test failed
		18 tests skipped
		1 test ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: IgnoreFixturesTest#test_ignores_parrots_fixtures (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList21_2) is available in the artifacts' activerecord log
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh activerecord
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #65297 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-21.1]
- #65157 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-20.2]
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3058191&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3058191&tab=artifacts#/activerecord) on master @ [cef0bd947590218ea6ac94f4f85dffc25e16fcd0](https://github.com/cockroachdb/cockroach/commits/cef0bd947590218ea6ac94f4f85dffc25e16fcd0):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/activerecord/run_1
	orm_helpers.go:245,orm_helpers.go:171,activerecord.go:229,activerecord.go:241,test_runner.go:801: 
		Tests run on Cockroach v21.2.0-alpha.00000000-1270-gcef0bd9475
		Tests run against activerecord 6.1
		6587 Total Tests Run
		6586 tests passed
		1 test failed
		18 tests skipped
		1 test ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: IgnoreFixturesTest#test_ignores_parrots_fixtures (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList21_2) is available in the artifacts' activerecord log
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh activerecord
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #65297 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-21.1]
- #65157 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-20.2]
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3063273&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3063273&tab=artifacts#/activerecord) on master @ [894a486398f12dc92b2f4af8c6acac4406b0d031](https://github.com/cockroachdb/cockroach/commits/894a486398f12dc92b2f4af8c6acac4406b0d031):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/activerecord/run_1
	orm_helpers.go:245,orm_helpers.go:171,activerecord.go:229,activerecord.go:241,test_runner.go:801: 
		Tests run on Cockroach v21.2.0-alpha.00000000-1331-g894a486398
		Tests run against activerecord 6.1
		6587 Total Tests Run
		6586 tests passed
		1 test failed
		18 tests skipped
		1 test ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
		--- FAIL: BasicsTest#test_default_values_are_deeply_dupped (unexpected)
		For a full summary look at the activerecord artifacts 
		
		An updated blocklist (activeRecordBlockList21_2) is available in the artifacts' activerecord log
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh activerecord
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #66219 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-66206 release-blocker]
- #65297 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-21.1]
- #65157 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-20.2]
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/activerecord.go
+++ b/pkg/cmd/roachtest/activerecord.go
@@ -21,7 +21,7 @@ import (
 var activerecordResultRegex = regexp.MustCompile(`^(?P<test>[^\s]+#[^\s]+) = (?P<timing>\d+\.\d+ s) = (?P<result>.)$`)
 var railsReleaseTagRegex = regexp.MustCompile(`^v(?P<major>\d+)\.(?P<minor>\d+)\.(?P<point>\d+)\.?(?P<subpoint>\d*)$`)
 var supportedRailsVersion = "6.1"
-var activerecordAdapterVersion = "v6.1.0"
+var activerecordAdapterVersion = "v6.1.2"
 
 // This test runs activerecord's full test suite against a single cockroach node.
 

--- a/pkg/cmd/roachtest/activerecord_blocklist.go
+++ b/pkg/cmd/roachtest/activerecord_blocklist.go
@@ -35,15 +35,21 @@ var activeRecordBlockList20_2 = blocklist{}
 
 var activeRecordIgnoreList21_2 = blocklist{
 	"ActiveRecord::CockroachDB::UnloggedTablesTest#test_gracefully_handles_temporary_tables": "modified to pass on 20.2",
-	"FixturesTest#test_create_fixtures": "flaky - FK constraint violated sometimes when loading all fixture data",
+	"FixturesTest#test_create_fixtures":                "flaky - FK constraint violated sometimes when loading all fixture data",
+	"IgnoreFixturesTest#test_ignores_books_fixtures":   "flaky - FK constraint violated sometimes when loading all fixture data",
+	"IgnoreFixturesTest#test_ignores_parrots_fixtures": "flaky - FK constraint violated sometimes when loading all fixture data",
 }
 
 var activeRecordIgnoreList21_1 = blocklist{
 	"ActiveRecord::CockroachDB::UnloggedTablesTest#test_gracefully_handles_temporary_tables": "modified to pass on 20.2",
-	"FixturesTest#test_create_fixtures": "flaky - FK constraint violated sometimes when loading all fixture data",
+	"FixturesTest#test_create_fixtures":                "flaky - FK constraint violated sometimes when loading all fixture data",
+	"IgnoreFixturesTest#test_ignores_books_fixtures":   "flaky - FK constraint violated sometimes when loading all fixture data",
+	"IgnoreFixturesTest#test_ignores_parrots_fixtures": "flaky - FK constraint violated sometimes when loading all fixture data",
 }
 
 var activeRecordIgnoreList20_2 = blocklist{
 	"ActiveRecord::CockroachDB::UnloggedTablesTest#test_gracefully_handles_temporary_tables": "modified to pass on 20.2",
-	"FixturesTest#test_create_fixtures": "flaky - FK constraint violated sometimes when loading all fixture data",
+	"FixturesTest#test_create_fixtures":                "flaky - FK constraint violated sometimes when loading all fixture data",
+	"IgnoreFixturesTest#test_ignores_books_fixtures":   "flaky - FK constraint violated sometimes when loading all fixture data",
+	"IgnoreFixturesTest#test_ignores_parrots_fixtures": "flaky - FK constraint violated sometimes when loading all fixture data",
 }

